Assets/Scripts/MazeGenerator.cs

```diff
@@ -0,0 +1,177 @@
+
+    private void Awake()
+    {
+        Instance = this;
+        GenerateMaze();
+    }
+
+    private void Start()
+    {
+
+    }
+
+    private void GenerateMaze()
+    {
+        Maze = new int[width, height];
+        for (int x = 0; x < width; x++)
+        {
+            for (int y = 0; y < height; y++)
+            {
+                Maze[x, y] = 1;
+            }
+        }
+        CurrentTile = Vector2.one;
+        _tiletoTry.Push(CurrentTile);
+        Maze = CreateMaze();
+        //GameObject ptype = null;
+
+        for (int i = 0; i <= Maze.GetUpperBound(0); i++)
+        {
+            for (int j = 0; j <= Maze.GetUpperBound(0); j++)
+            {
+                if (Maze[i, j] == 1)
+                {
+                    //ptype = GameObject.CreatePrimitive(PrimitiveType.Cube);
+                    //ptype.transform.position = new Vector3(i * ptype.transform.localScale.x, j * ptype.transform.localScale.y, 0);
+                    //if (brick != null)
+                    //{
+                    //    ptype.renderer.material = brick;
+                    //}
+                    //ptype.transform.parent = transform;
+
+                    Instantiate(Brick, new Vector3(i, 0, j), Quaternion.identity);
+                }
+                else if (Maze[i, j] == 0)
+                {
+                    pathMazes.Add(new Vector3(i, 0, j));
+                }
+            }
+        }
+    }
+
+    public int[,] CreateMaze()
+    {
+        //local variable to store neighbors to current square
+        List<Vector2> neighbors;
+        //as long as there are sitll tiles to try
+        while (_tiletoTry.Count > 0)
+        {
+            //excavate the square we are on
+            Maze[(int)CurrentTile.x, (int)CurrentTile.y] = 0;
+
+            //get all valid neighbors for the new tile
+            neighbors = GetValidNeighbors(CurrentTile);
+
+            //if there are any interesting looking neighbors
+            if (neighbors.Count > 0)
+            {
+                //remember this tile by pushing it onto the stack
+                _tiletoTry.Push(CurrentTile);
+                //move on to a random of the neighboring tiles
+                CurrentTile = neighbors[rnd.Next(neighbors.Count)];
+            }
+            else
+            {
+                //if there were no neighbors to try, it's a dead end
+                //toss the tile out
+                //(and return to a previous tile on the list to check)
+                CurrentTile = _tiletoTry.Pop();
+            }
+        }
+
+        return Maze;
+    }
+
+    /// <summary>
+    /// Get all the prospective neighboring tiles
+    /// </summary>
+    /// <param name="centerTile">The tile to test</param>
+    /// <returns>Any and all valid neighbors</returns>
+    private List<Vector2> GetValidNeighbors(Vector2 centerTile)
+    {
+
+        List<Vector2> validNeighbors = new List<Vector2>();
+
+        foreach (var offset in offsets)
```

Comment:
`var` is gross and makes code more difficult to read. Instead make it explicit so others can read it easier